### PR TITLE
[W5.11][W15-A1]Jason Leo

### PR DESCRIPTION
**Pull request by jas5469**
I have added 2 features. 1 is the sorting function and 1 is the case insensitivity in find function

### CODE REVIEW

File: src/seedu/addressbook/commands/FindCommand.java [hunk 1]
```diff
@@ -48,13 +49,34 @@ public CommandResult execute() {
...
         return matchedPersons;
     }
 
+    /**
+     * Return keyword in lower case so as to make it case insensitive
+     * @param keyToChange contains keyword to be converted to lower case
+     * @return keywords into lower case
+     */
+
+    private static Set<String> changeKeywordsintoLowerCase(Set<String> keyToChange){
```
Comment (MuhdNurKamal) [inline src/seedu/addressbook/commands/FindCommand.java:69]: Instead of creating 2 new Hashsets, I think it's clearer to create 1 new set, then transfer each word from keyToChange after converting it to lower case. then return the new set

Comment (MuhdNurKamal) [reply src/seedu/addressbook/commands/FindCommand.java:69]: An method name shouldn't be changeKeywordsintoLowerCase because it doesnt change keywords but rather return a new set of keywords

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.*;
+
+public class SortCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sorted the address book in lexicographical order.\n"
```
Comment (MuhdNurKamal) [inline src/seedu/addressbook/commands/SortCommand.java:12]: The command doesnt actually sort the address book, rather it only diplays the address book in sorted order(The internal representation is still not sorted)

File: src/seedu/addressbook/data/person/Person.java [hunk 3]
```diff
@@ -84,4 +84,9 @@ public String toString() {
         return getAsTextShowAll();
     }
 
+    @Override
+    public int compareTo(Person personToCompare){
```
Comment (MuhdNurKamal) [inline src/seedu/addressbook/data/person/Person.java:88]: Perhaps naming personToCompare as other or otherPerson is clearer

File: src/seedu/addressbook/commands/SortCommand.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    /**
+     * Sort the list in lexicographical order using names
+     *
+     *
+     * @return list of supported persons
+     */
+    private List<ReadOnlyPerson> sortPersonLists (List<ReadOnlyPerson> unsortedList){
+        List<ReadOnlyPerson> sortedListOfPersons = new ArrayList<ReadOnlyPerson>(unsortedList);
+        Collections.sort(sortedListOfPersons, new Comparator<ReadOnlyPerson>() {
```
Comment (MuhdNurKamal) [inline src/seedu/addressbook/commands/SortCommand.java]: Why use an anonymous Comparator which compares name when you could just use the compareTo method you have overidden in person class(You could make ReadOnlyPerson extend Comparable)

**Comment by MuhdNurKamal** [comment]
Overall good effort in implementing both features.

File: src/seedu/addressbook/commands/FindCommand.java [hunk 5]
```diff
@@ -5,19 +5,20 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.*;
```
Comment (devamanyu) [inline src/seedu/addressbook/commands/FindCommand.java:8]: Avoid auto-formatting unrelated parts of the code. Doing so can result in unrelated changes being included in the commit. You can use Sourcetree to look through changes before committing, and discard any unrelated changes.

File: src/seedu/addressbook/commands/FindCommand.java (hunk 1 above)
Comment (devamanyu) [reply src/seedu/addressbook/commands/FindCommand.java:69]: These are valid points. Although we are not enforcing space optimization, but its always good to use minimal space while implementing.

File: src/seedu/addressbook/commands/FindCommand.java [hunk 6]
```diff
@@ -48,13 +49,34 @@ public CommandResult execute() {
...
+     */
+
+    private static Set<String> changeKeywordsintoLowerCase(Set<String> keyToChange){
+        Set<String> removeStringSet = new HashSet<>();
+        Set<String> addStringSet = new HashSet<>();
+
+        for(String string : keyToChange){
+            removeStringSet.add(string);
+            addStringSet.add(string.toLowerCase());
+        }
```
Comment (devamanyu) [inline src/seedu/addressbook/commands/FindCommand.java:76]: Scope for SLAP here. In the current logic, too much is happening in this method. Try to use slap to make the code self-readable.

**Review by devamanyu — COMMENTED** [review]
This overall a good implementation. The feedback from Nur and the following commits have alleviated the significant lackings. Please close this PR after reading the comments.
